svm/src/transaction_processor.rs

```diff
@@ -462,7 +445,15 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         lamports_per_signature: u64,
         rent_collector: &RentCollector,
         error_counters: &mut TransactionErrorMetrics,
-    ) -> transaction::Result<(FeeDetails, AccountSharedData, u64)> {
+    ) -> transaction::Result<(ComputeBudgetLimits, FeeDetails, AccountSharedData, u64)> {
+        let compute_budget_limits = process_compute_budget_instructions(
+            message.program_instructions_iter(),
+        )
+        .map_err(|err| {
+            error_counters.invalid_compute_budget += 1;
+            err
+        })?;
```

Comment:
note that we now fail with an error immediately here in `validate_transaction_fee_payer` if compute budget processing fails rather than waiting until execution to fail. Both before and after this refactoring we mark such a transaction as invalid and don't execute it so this won't cause consensus issues.